test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/ObjectMethodExecutorTest.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Internal;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Core.Internal
+{
+    public class ObjectMethodExecutorTest
+    {
+        private TestObject _targetObject = new TestObject();
+        private TypeInfo targetTypeInfo = typeof(TestObject).GetTypeInfo();
+
+        [Fact]
+        public void ExecuteStaticVoidMethod()
+        {
+            var executor = GetExecutorForMethod("StaticVoidMethod");
+            var result = executor.Execute(null, null);
+            Assert.Same(null, result);
+        }
+
+        [Fact]
+        public void ExecuteStaticValueMethod()
+        {
+            var executor = GetExecutorForMethod("StaticValueMethod");
+            var result = executor.Execute(null, new object[] { 10 });
+            Assert.Equal(10, (int)result);
+        }
+
+        [Fact]
+        public void ExecuteStaticMethodThrowsException()
+        {
+            var executor = GetExecutorForMethod("StaticMethodThrowsException");
+            Assert.Throws<NotImplementedException>(
+                        () => executor.Execute(null, new object[] { 10 }));
+            
+        }
+
+        [Fact]
+        public void ExecuteValueMethod()
+        {
+            var executor = GetExecutorForMethod("ValueMethod");
+            var result = executor.Execute(_targetObject, new object[] { 10 , 20 });
+            Assert.Equal(30, (int)result);
+        }
+
+        [Fact]
+        public void ExecuteVoidValueMethod()
+        {
+            var executor = GetExecutorForMethod("VoidValueMethod");
+            var result = executor.Execute(_targetObject, new object[] { 10 });
+            Assert.Same(null, result);
+        }
+
+        [Fact]
+        public void ExecuteValueMethodWithReturnType()
+        {
+            var executor = GetExecutorForMethod("ValueMethodWithReturnType");
+            var result = executor.Execute(_targetObject, new object[] { 10 });
+            Assert.True(result is TestObject);
```

Comment:
Use `Assert.IsType<>` here.
```
var resultObject = Assert.IsType<TestObject>(result);
```

Any time you're using `Assert.True` or `Assert.False` think carefully, there's almost always something better.